packages/client-node/__tests__/integration/node_exec.test.ts

```diff
@@ -66,4 +68,33 @@ describe('[Node.js] exec result streaming', () => {
       expect(await rs.json()).toEqual([{ number: '0' }])
     })
   })
+
+  describe('custom insert streaming with exec', () => {
```

Comment:
can we add a test that it doesn't blow with an empty stream, closed stream, and other edge cases?